R/utils.R

```diff
@@ -168,3 +168,17 @@ unit_vector <- function(c1, c2){
 }
 
 .intersectionSize <- function(...) { length(intersect(...))}
+
+dfs_traverse <- function(df, id) {
```

Comment:
better use functions from igraph package, instead of reimplementing dfs, e.g. you can create graph from the table and calculate distances from the root vertex to all other